### PR TITLE
[ZEPPELIN-5986] Bump Maven surefire/failsafe plugins to recover JUnit5 tests

### PR DESCRIPTION
**Pull request by pan3793**
### What is this PR for?

This is the third attemps of ZEPPELIN-5986, partial of this PR are credited to @Reamer, and I also got help from @zjffdu offline.

- https://github.com/apache/zeppelin/pull/4699 (first attempt)
- https://github.com/apache/zeppelin/pull/4725 (second attempt)

#### Background

I found that some tests are skipped while tackling the Hadoop3 upgrading, and eventually I realized that was caused by the low version of maven-failsafe-plugin not supporting JUnit5 annotations.

Actually, maven-surefire-plugin(for UT) and maven-failsafe-plugin(for IT) are developed in the same [project](https://github.com/apache/maven-surefire) and they share the same release cycle thus have same versions.

#### Changes of this PR

1. upgrade these two plugins to the latest 3.2.2 to recover the IT written in JUnit5
2. disable the broken Livy IT, see discussion in https://github.com/apache/zeppelin/pull/4725
3. fix the broken `SparkIntegrationTest`, thanks @Reamer 
4. fix the broken Selenium Tests
5. enhance `WebDriverManager`, to allow specify web browser for Selenium integration tests via `export ZEPPELIN_SELENIUM_BROWSER=[Chrome|Firefox|Safari]`

### What type of PR is it?

Test.

### Todos

* [ ] - Livy integration tests were broken, and is skipped in this PR.

### What is the Jira issue?

ZEPPELIN-5986

### How should this be tested?

Pass GHA.

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No.

### CODE REVIEW

**Comment by pan3793** [comment]
@Reamer I figured out why frontend IT failed. Actually, the changes in ZEPPELIN-5848 have some issues, but due to the lower version of the Maven plugin, all JUnit5 tests are skipped.

The real matter is, `AaaIT` creates an instance `BbbIT` in the test cases, and calls `BbbIT#method`, obviously, `BbbIT#beforeClass` won't be called, so that the member variables initialization is skipped, then NPE occurs, I fixed it with minimal changes, the first round CI result looks good (only one case failed, may be a flaky case, let me try another round)

**Comment by pan3793** [comment]
also cc @jongyoul and @zjffdu

File: zeppelin-integration/src/test/java/org/apache/zeppelin/AbstractZeppelinIT.java [hunk 1]
```diff
@@ -118,17 +118,7 @@ protected boolean waitForText(final String txt, final By locator) {
...
-        .pollingEvery(Duration.of(1, ChronoUnit.SECONDS))
-        .ignoring(NoSuchElementException.class);
-
-    return wait.until(new Function<WebDriver, WebElement>() {
-      @Override
-      public WebElement apply(WebDriver driver) {
-        return driver.findElement(locator);
-      }
-    });
+    return ZeppelinITUtils.pollingWait(manager.getWebDriver(), locator, timeWait);
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/AbstractZeppelinIT.java:121]: Why did you outsource this function?

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java [hunk 2]
```diff
@@ -58,4 +69,58 @@ public static void turnOnImplicitWaits(WebDriver driver) {
...
+
+    return wait.until(new Function<WebDriver, WebElement>() {
+      @Override
+      public WebElement apply(WebDriver driver) {
+        return driver.findElement(locator);
+      }
+    });
+  }
+
+  public static void authenticationUser(WebDriver driver, String userName, String password) {
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java:87]: You could also move this function to `AbstractZeppelinIT`.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java [hunk 3]
```diff
@@ -58,4 +69,58 @@ public static void turnOnImplicitWaits(WebDriver driver) {
...
+        MAX_BROWSER_TIMEOUT_SEC).sendKeys(password);
+    ZeppelinITUtils.pollingWait(
+        driver,
+        By.xpath("//*[@id='loginModalContent']//button[contains(.,'Login')]"),
+        MAX_BROWSER_TIMEOUT_SEC).click();
+
+    ZeppelinITUtils.sleep(1000, false);
+  }
+
+  public static void logoutUser(WebDriver driver, String userName) throws URISyntaxException {
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java:111]: You could also move this function to `AbstractZeppelinIT`.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java [hunk 4]
```diff
@@ -184,8 +184,7 @@ private void setPythonParagraph(int num, String text) {
   void testGloballyAction() throws Exception {
     try {
       //step 1: (admin) login, set 'globally in shared' mode of python interpreter, logout
-      InterpreterModeActionsIT interpreterModeActionsIT = new InterpreterModeActionsIT();
-      interpreterModeActionsIT.authenticationUser("admin", "password1");
+      authenticationUser("admin", "password1");
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java:187]: This class also contains the method `authenticationUser`, we should only have the function once. My suggestion: not static in `AbstractZeppelinIT`.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java [hunk 5]
```diff
@@ -203,12 +202,12 @@ void testGloballyAction() throws Exception {
       clickAndWait(By.xpath(
           "//div[@class='modal-dialog']//div[@class='bootstrap-dialog-footer-buttons']//button[contains(., 'OK')]"));
       clickAndWait(By.xpath("//a[@class='navbar-brand navbar-title'][contains(@href, '#/')]"));
-      interpreterModeActionsIT.logoutUser("admin");
+      logoutUser("admin");
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java:205]: This class also contains the method `logoutUser`, we should only have the function once. My suggestion: not static in `AbstractZeppelinIT`.

**Review by Reamer — COMMENTED** [review]
Great that you were able to get the tests running. I have added a few comments.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java (hunk 2 above)
Comment (pan3793) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java]: adopted

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java (hunk 3 above)
Comment (pan3793) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java]: adopted

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java (hunk 4 above)
Comment (pan3793) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java:152]: adopted

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java (hunk 5 above)
Comment (pan3793) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/integration/InterpreterModeActionsIT.java:170]: adopted

**Comment by pan3793** [comment]
@Reamer comments are addressed, please take a look again

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java [hunk 6]
```diff
@@ -18,11 +18,22 @@
 package org.apache.zeppelin;
 
 
+import com.google.common.base.Function;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
```
Comment (Reamer) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java:24]: Can you please clean up the imports.

**Review by Reamer — CHANGES_REQUESTED** [review]
Almost perfect. Just import cleanup.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java (hunk 6 above)
Comment (pan3793) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java]: done

**Review by Reamer — APPROVED** [review]
LGTM :+1:

**Comment by pan3793** [comment]
@Reamer looks like the idea is removing the leading `zeppelin/` prefix on finding the Python interpreter process, it should help if the developer's local git repo folder is not "zeppelin", but I suppose it does not help to address the flaky test.

I just push a empty commit to trigger the CI again

**Pull request merged**